pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -166,7 +168,7 @@ func (nim *nodeInfoManager) updateNode(updateFuncs ...nodeUpdateFunc) error {
 		}
 
 		if needUpdate {
-			_, updateErr := nodeClient.Update(node)
+			_, _, updateErr := nodeutil.PatchNodeStatus(kubeClient.CoreV1(), types.NodeName(node.Name), originalNode, node)
```

Comment:
It's not always the node status that gets updated. For example, for labels are added to the Node object for topology.

Dumb question: why wasn't it working before?

Reply:
Previously the fields are updated in the fetched Node object directly, and the object is then passed to `Update()`. So `Update()` still sent the request with updated fields right? Why weren't allocatable/capacity updated?

Reply:
Added a node about why we are using `PatchNodeStatus` here.